app/src/test/java/com/keylesspalace/tusky/SpanUtilsTest.kt

```diff
@@ -1,182 +1,187 @@
 package com.keylesspalace.tusky
 
 import android.text.Spannable
+import com.keylesspalace.tusky.util.FoundMatchType
+import com.keylesspalace.tusky.util.MENTION_PATTERN_STRING
+import com.keylesspalace.tusky.util.PatternFinder
+import com.keylesspalace.tusky.util.TAG_PATTERN_STRING
 import com.keylesspalace.tusky.util.highlightSpans
-import org.junit.Assert
+import com.keylesspalace.tusky.util.twittertext.Regex
+import java.util.regex.Pattern
+import org.junit.Assert.assertEquals
 import org.junit.Test
 import org.junit.runner.RunWith
 import org.junit.runners.Parameterized
 
-class SpanUtilsTest {
-    @Test
-    fun matchesMixedSpans() {
-        val input = "one #one two: @two three : https://thr.ee/meh?foo=bar&wat=@at#hmm four #four five @five ろく#six"
-        val inputSpannable = FakeSpannable(input)
-        highlightSpans(inputSpannable, 0xffffff)
-        val spans = inputSpannable.spans
-        Assert.assertEquals(6, spans.size)
+/** The [Pattern.UNICODE_CHARACTER_CLASS] flag is not supported on Android, on Android it is just always on.
+ * Since thesse tests run on a regular Jvm, we need a to set this flag or they would behave differently.
+ * */
```

Comment:
This is sooo annoying. Why can't the flag just be ignored on Android, no it crashes 🙄